templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py

```diff
@@ -0,0 +1,419 @@
+    genericAttachments = {'version': '1','contentType': 'application/vnd.amazonaws.card.generic','genericAttachments':[{"title":"response buttons","buttons":choicelist}]}
+    event['res']['session']['appContext']['responseCard'] = genericAttachments
+    event['res']['session']['appContext']['altMessages']['markdown'] = "Please select one of the options below:"
+
+    intCounter = 0
+    strChoiceList = ""
+    for items in choicelist: 
+        if strChoiceList != '': 
+            strChoiceList = strChoiceList + ", "
+        strChoiceList = strChoiceList + choicelist[intCounter]['text']
+        intCounter = intCounter + 1
+    event['res']['session']['appContext']['altMessages']['ssml'] = get_SSML_output("Please select one of these options: " + strChoiceList)
+
+    return event
+
+
+"""
+function: query_enrollments_for_student
+This function retrieves students' active enrollments 
+"""
+def query_enrollments_for_student(event, canvas, student_user_name):
+    # Get the user using user_id to match with LMS SIS_ID
+    user = getCanvasUser (canvas, student_user_name)
+
+    if user:
+        courses = user.get_courses(enrollment_status='active',include=['syllabus_body'])
+        # Loop through the courses.
+        course_names = [course.name for course in courses]
+
+    result = {"CourseNames": course_names}
+
+    return_courses = result['CourseNames']
+    if return_courses:
+        choicelist = []
+        for i in return_courses:
+            choicelist.append({'text':i,'value':"more information about my {} course".format(i)})
+        genericAttachments = {'version': '1','contentType': 'application/vnd.amazonaws.card.generic','genericAttachments':[{"title":"response buttons","buttons":choicelist}]}
+        event['res']['session']['appContext']['responseCard'] = genericAttachments
+        event['res']['session']['appContext']['altMessages']['markdown'] = "Please select one of the options below:"
+
+        intCounter = 0
+        strChoiceList = ""
+        for items in choicelist: 
+            if strChoiceList != '': 
+                strChoiceList = strChoiceList + ", "
+            strChoiceList = strChoiceList + choicelist[intCounter]['text']
+            intCounter = intCounter + 1
+        event['res']['session']['appContext']['altMessages']['ssml'] = get_SSML_output("Please select one of these options: " + strChoiceList)
+    else:
+        return_message = "You are not currently enrolled in any courses."
+        set_alt_message (event, return_message)
+
+    return event
+    
+
+"""
+function: query_courses_for_student
+This function retrieves course options across all active enrolled courses, or for a particular course, for the student
+for example: more information about {course name}
+"""
```

Comment:
please move function documentation inside the function (multiple occurrences).
Example:
```
def query_courses_for_student(event, canvas, student_user_name, course_name_fo_filter):
    """
    function: query_courses_for_student
    This function retrieves course options across all active enrolled courses, or for a particular course, for the student
    for example: more information about {course name}
    """
    ....
```

Reply:
Addressed this in latest commit. 
Commit: https://github.com/aws-solutions/aws-qnabot/pull/461/commits/4d8b35beb4301f1a8b7736f8a3eacc8638164882